.ci/build.sh

```diff
@@ -118,7 +125,11 @@ function archive {
 }
 
 if declare -F "$1" > /dev/null; then
-    $1
+    if [ -n "${2-}" ]; then
```

Comment:
Can you explain what this is for?

Reply:
On [line 25](https://github.com/opensearch-project/opensearch-benchmark/blob/main/.ci/build.sh#L25) `set -u` is run which causes the script to throw an error when a variable is referenced but unset. The unit tests are run with `./build.sh build` whereas the integ tests are run with `./build build_it <py version>`. So for unit tests, `$2` is unset and throws an exception.

`${2-}` gets the value of $2 if set, otherwise returns an empty value. `-n` checks if the provided value is non-null/empty/0.

It's essentially a hacky way to check if a `$2` parameter is set without throwing an exception. This file should be holistically re-evaluated IMO as it's a bit messy (for example, we run the opposite of `set -u` [in](https://github.com/opensearch-project/opensearch-benchmark/blob/main/.ci/build.sh#L91-L92) [multiple](https://github.com/opensearch-project/opensearch-benchmark/blob/main/.ci/build.sh#L39-L40) [functions](https://github.com/opensearch-project/opensearch-benchmark/blob/main/.ci/build.sh#L64-L65)).

Reply:
You can replace both the cases with a single `"$@"` and eliminate the `if` statement.

Reply:
That is much cleaner and scales without modifications if we were to add a second parameter to one of the functions. Will change, thanks!